src/sink.cc

```diff
@@ -164,9 +165,36 @@ void Sink::Tick() {
   using std::vector;
   LOG(cyclus::LEV_INFO3, "SnkFac") << prototype() << " is ticking {";
 
-  double requestAmt = RequestAmt();
+  double amt = RequestAmt();
+  double requestAmt = 0;
+
+  LOG(cyclus::LEV_INFO3, "SnkFac") << prototype() << " has default request amount " << amt;
+
   // inform the simulation about what the sink facility will be requesting
-  if (requestAmt > cyclus::eps()) {
+  if (amt > cyclus::eps()) {
+    if (random_size == "None") {
+      requestAmt = amt;
+    }
+    else if (random_size == "UniformInt") {
+      requestAmt = context()->random_uniform_int(0, amt);
+    }
+    else if (random_size == "UniformReal") {
+      requestAmt = context()->random_uniform_real(0, amt);
+    }
+    else if (random_size == "NormalReal") {
+      requestAmt = context()->random_normal_real(amt * random_size_mean,
```

Comment:
Just to make sure I understand, this will be sampling from only the lower half of the Normal distribution, right?

Reply:
No, the user also defines a mean and standard deviation as a fraction of the maximum value. You could sample from the lower half by setting &mu; = 1.0.

Example
![truncated_normal (2)](https://github.com/cyclus/cycamore/assets/8053456/0ccc60e0-a0fd-4347-97ab-0e8227ad2b1a)

Since sink is generally dominated by its throughput (capacity) until max inventory is almost reached, this implementation could also take a fixed mean and standard deviation in kg, rather than fractional. It would get wonky at the end if remaining inventory is small and mean is large (would probably have to automatically adjust the distribution to prevent huge numbers of random numbers being sampled and rejected) 

I could personally imagine myself using either way, not sure if you have a strong preference

Reply:
Got it... I was stuck thinking about the test case where mean = 1, and didn't think through the case when mean < 1.  This makes sense.